games-roguelike/moria: version 5.7.10

steils opened the pull request:
Closes: https://bugs.gentoo.org/674540
Package-Manager: Portage-2.3.57, Repoman-2.3.12
Signed-off-by: Stefan Strogin <stefan.strogin@gmail.com>

juippis commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,47 @@
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit eutils cmake-utils
```
Do you need eutils here? I see you use newbin instead of make_wrapper anyway.

steils replied:
No I don't. Forgot to remove. Thanks.

juippis commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,47 @@
+RDEPEND=">=sys-libs/ncurses-6:0"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig
+	dev-util/cmake
```
Doesn't cmake-utils take care of this?

steils replied:
Thanks, I didn't know that. I've looked into cmake-utils.eclass, it does take care of BDEPEND.
Should we not set BDEPEND explicitly if it is set via eclass? I haven't found anything about that in devmanual.

juippis replied:
IMHO it's just repetition, and quite obvious that these build-system eclasses will provide the build-tool for DEPEND. But I mentioned it because @Zlogene always says to remove them, unless specific version is needed :)

juippis commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,47 @@
+	dev-util/cmake
+"
+
+S=${WORKDIR}/umoria-${PV}
```
`S=" "` please, same for BUILD_DIR.

steils commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,45 @@
+}
+
+src_compile() {
+	cmake-utils_src_compile all
```
You are right once again. Thanks for the help!

Zlogene commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,41 @@
+
+src_configure() {
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr/share"
```
why do you want to redefine eclass-determined option? The whole `src_configure()` is useless here

steils replied:
Fisrt of all `umoria` needs all these files: https://github.com/dungeons-of-moria/umoria/blob/master/src/config.cpp#L14
In the directory with the binary.
I.e. the binary is placed in `/usr/bin/umoria`, then the program will try to open `/usr/bin/data/splash.txt`, `/usr/bin/data/welcome.txt`, `/usr/bin/LICENSE` and so on.

So I place the binary into `/usr/share/umoria/` and use a wrapper to copy the binary and all these data files into `~/.local/share/umoria/`. This is the solution used in `games-roguelike/tomenet` (BTW, I have made a PR for it https://github.com/gentoo/gentoo/pull/10856)

Another solution is to patch `src/config.cpp` to make the program look for these files in more Unix-traditional directories. It's the solution used in Debian: https://sources.debian.org/src/moria/5.7.10+20181022-2/src/config.cpp/#L14. I can remake it in this way if you think it's more correct and it's worth it.

steils replied:
> So I place the binary into `/usr/share/umoria/`

I mean that's why I need `-DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr/share"`

chewi replied:
I would prefer a patch and/or sed or if it's really awkward, put it in `/opt`. We try to keep `/usr/share` platform-neutral as much as possible.

steils replied:
@chewi Thank you for the comment. I'll remake the ebuild then.

Will it be acceptable to put `LICENSE` file into `/usr/share/doc/${P}` or anywhere else?
If we don't install `LICENSE` at all, the game will show an error when it will try to open it, but it won't affect the game itself.

Zlogene commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,41 @@
+BDEPEND="virtual/pkgconfig"
+
+S="${WORKDIR}/umoria-${PV}"
+BUILD_DIR="${S}"
```
this is useless

steils replied:
Unfortunately the game's build instructions order to run cmake from the source top directory. It fails if you run it in `${WORKDIR}/${P}_build`

chewi replied:
Do this instead.

`CMAKE_IN_SOURCE_BUILD=true`

steils replied:
@chewi Thanks!

Zlogene commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,41 @@
+
+src_install() {
+	cmake-utils_src_install
+	dodoc README.md docs/*
```
you do not need globs installing dirs, use `dodoc -r`

steils replied:
Sure. Thanks for the hint!

Zlogene commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,41 @@
+	doins data/*.txt
+
+	insinto /usr/share/umoria
+	doins LICENSE data/scores.dat
```
we do not install LICENSE files

steils replied:
I know, I had to. The binary tries to open it, see: https://github.com/gentoo/gentoo/pull/10899#discussion_r251069710

steils commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,41 @@
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit cmake-utils
```
```
EAPI=7

inherit cmake-utils
```
ok

steils commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,41 @@
+KEYWORDS="~amd64 ~x86 ~x86-fbsd"
+IUSE=""
+
+RDEPEND=">=sys-libs/ncurses-6:0"
```
Thanks.
Umoria needs ncurses-6.x. Is `>=sys-libs/ncurses-6.0:0=` correct?

chewi replied:
Yep.

steils commented:
@juippis @Zlogene @chewi, thank you all for the review! I have sent the next version. Here are changes:

- Removed `src_configure` and rewrote `src_install` to install game data and doc files into proper directories accdoring to FHS. Hope I've done it right.
- Removed the wrapper. Added a patch and a sed command to make the binary look for its files in the correct directories.
- Used `CMAKE_IN_SOURCE_BUILD`.
- Correctly slotted ncurses.
- Added `moria.6` manual page from an old Umoria version.
- Updated `DESCRIPTION`, I believe it was incorrect.

juippis commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,49 @@
+
+	insinto /var/lib/moria
+	doins data/scores.dat
+	chgrp games "${ED}"/var/lib/moria/scores.dat
```
Would fowners work here? If not, append with `|| die` please. (No need to answer, but I was just wondering what happens if games group doesn't exist when using chgrp?)

steils replied:
Changed to fowners and fperms. Thanks!

juippis commented on games-roguelike/moria/moria-5.7.10.ebuild:
```diff
@@ -0,0 +1,49 @@
+	insinto /var/lib/moria
+	doins data/scores.dat
+	chgrp games "${ED}"/var/lib/moria/scores.dat
+	chmod g+w "${ED}"/var/lib/moria/scores.dat
```
Use fperms if possible, and if not, append with `|| die`.
